rules/rollup.go

```diff
@@ -105,15 +73,46 @@ func rollupTargetViewsToTargets(views []RollupTargetView) []RollupTarget {
 	return targets
 }
 
+// RollupTarget dictates how to roll up metrics. Metrics associated with a rollup
+// target will be grouped and rolled up across the provided set of tags, named
+// with the provided name, and aggregated and retained under the provided policies.
+type RollupTarget struct {
+	Name     []byte
+	Tags     [][]byte
+	Policies []policy.Policy
+}
+
+func newRollupTarget(target *schema.RollupTarget) (RollupTarget, error) {
+	if target == nil {
+		return emptyRollupTarget, errNilRollupTargetSchema
+	}
+	policies, err := policy.NewPoliciesFromSchema(target.Policies)
+	if err != nil {
+		return emptyRollupTarget, err
+	}
+	tags := make([]string, len(target.Tags))
+	copy(tags, target.Tags)
+	sort.Strings(tags)
+	return RollupTarget{
+		Name:     []byte(target.Name),
+		Tags:     bytesArrayFromStringArray(tags),
+		Policies: policies,
+	}, nil
+}
+
 func (t *RollupTarget) sameTransform(other RollupTarget) bool {
 	if !bytes.Equal(t.Name, other.Name) {
 		return false
 	}
 	if len(t.Tags) != len(other.Tags) {
 		return false
 	}
-	for i := 0; i < len(t.Tags); i++ {
-		if !bytes.Equal(t.Tags[i], other.Tags[i]) {
+	clonedTags := stringArrayFromBytesArray(t.Tags)
```

Comment:
could sort the byte slices as in this [example](https://gist.github.com/schmohlio/615ab4d47bc01020786ef58aec622fdf) to avoid the extra allocs, if this isn't in the hot path though this seems simpler to me

Reply:
Yeah I'll have to alloc unfortunately to avoid mutating the original tags slice in the rollup targets. I could have avoided the byte slice to string conversion yes, but this seems to be more readable.